docs/book/advanced-guide/gcp_entrypoint.md

```diff
@@ -0,0 +1,277 @@
+
+In order to use the container-registry at a later point you will need to 
+set the container registry uri. This is how it is usually constructed:
+
+```bash
+export CONTAINER_REGISTRY_URI=eu.gcr.io/<PROJECT_NAME> 
+```
+
+{% hint style="info" %}
+The container registry has four options: **gcr**.io , us.gcr.io , eu.gcr.io , 
+or asia.gcr.io. Choose the one appropriate for you. 
+{% endhint %}
+
+## Step 6/10 Set up Cloud Storage as Artifact Store
+
+Storing of step artifacts is an important part of reproducible MLOps. 
+[create](https://console.cloud.google.com/storage/create-bucket)
+
+Within the configuration of the newly created bucket you can find the 
+gsutil URI which you will need at a later point.
+```bash
+export GSUTIL_URI=...
+```
+
+## Step 7/10 Set up a Cloud SQL instance as Metadata Store
+
+One of the most complex resources that you'll need to manage is the MySQL
+database. 
+
+To start, we [create](https://console.cloud.google.com/sql/instances/create;engine=MySQLe)
+a MySQL database. Once created, it will take some time for the database to be 
+set up. 
+
+Once it is set up you can find the IP-address. The password you set during 
+creation of the instance is the root password. The default port for mysql is 
+3306. 
+
+```bash
+export DB_HOST_IP=...
+export DB_PORT=3306
+export DB_USER=root
+export DB_PWD=...
+```
+
+Time to set up the connections to our database. To do this you'll need to go 
+into the `Connections` menu. Under the `Networking` tab you'll need to add 
+**0.0.0.0** to the authorized networks, thereby allowing all incoming traffic 
+from everywhere. (Feel free to restrict this to your outgoing IP address)
+
+For security reasons, it is also recommended to configure your database to only 
+accept SSL connections. You'll find the relevant setting in the **Security*+ 
+tab. Select **SSL Connections only** in order to encrypt all traffic with your 
+database.
+
+Now **Create Client Certificate** and download all three files. Export the paths 
+to these three files as follows with a leading **@**.
+```bash
+export SSL_CA=@</PATH/TO/DOWNLOADED/SERVER-CERT>
+export SSL_CERT=@</PATH/TO/DOWNLOADED/CLIENT-CERT>
+export SSL_KEY=@</PATH/TO/DOWNLOADED/CLIENT-KEY>
+```
+
+Finally, head on over to the `Databases` submenu and create your database and
+export its name. 
+
+```bash
+export DB_NAME=...
+```
+
+## Step 8/10 - Create a Service Account
+
+All the resources are created. Now we need to make sure the instance performing 
+the compute engine (Vertex AI) needs to have the relevant permissions to access 
+the other resources. For this you'll need to go
+[here](https://console.cloud.google.com/iam-admin/serviceaccounts/create) to 
+create a service account. Give it a relevant name and allow access to the
+following roles:
+
+* Vertex AI Custom Code Service Agent
+* Vertex AI Service Agent
+* Container Registry Service Agent
+* Secret Manager Admin
```

Comment:
There is a way to do this conveniently in the command line which we could paste  here 

```
gcloud projects add-iam-policy-binding PROJECT_ID \
    --member="serviceAccount:SERVICE_ACCOUNT_ID@PROJECT_ID.iam.gserviceaccount.com" \
    --role="ROLE_NAME"
```